### ISSUE TITLE
Huge freeze when using PageStorageKey and big ListView

### ISSUE THREAD

**Issue by nt4f04uNd** [flutter/flutter#41027]
## Details

On a main route I have a ListView of 300 elements in which every tile represents a track on user device and then I have a route that shows user the details about track/allows to seek/pause/click next/click prev and so on. 

When I switch between these routes I really want to preserve ListView scroll and the only suggested solution is to use `PageStorageKey`.

But the more I scroll down my ListView, the more perfomance of `Navigator.of(context).pop()` goes down, before it reaches the stage, when whole route transition takes for like 0.5 seconds and it performs without any animation, just blinks.

## Steps to reproduce

1. Have a complex and long list
2. Scroll down to its bottom
3. Put `PageStorageKey` key to a `List.builder`
4. Try to switch route and go back

## Also see my application [repository issue branch](https://github.com/nt4f04uNd/flutter_music_player/tree/pagestoragekey-perfomance-issue)
(excuse me, but it's too hard for me to create separate repo only for report)

### `flutter analyze`

```shell
c:\dev\projects\flutter_music_player\app>flutter analyze
Analyzing app...


   info - Await only futures - lib\components\animatedPlayPauseButton.dart:44:5 - await_only_futures
   info - Unused import: 'package:app/heroes/albumArtHero.dart' - lib\components\bottomTrackPanel.dart:1:8 -
          unused_import
   info - This class (or a class which this class inherits from) is marked as '@immutable', but one or more of its
          instance fields are not final: _AlbumArtPicHero.horizontalMargin - lib\heroes\albumArtHero.dart:23:7 -
          must_be_immutable
   info - The value of the local variable 'widgetWidth' isn't used - lib\marquee.dart:49:9 - unused_local_variable
   info - The value of the local variable 'widgetHeight' isn't used - lib\marquee.dart:50:12 - unused_local_variable
   info - Cancel instances of dart.async.StreamSubscription - lib\player\player.dart:64:32 - cancel_subscriptions
   info - The value of the field '_durationChangeListenerSubscription' isn't used - lib\player\player.dart:64:32 -
          unused_field
   info - Cancel instances of dart.async.StreamSubscription - lib\player\player.dart:67:22 - cancel_subscriptions
   info - The value of the field '_eventSubscription' isn't used - lib\player\player.dart:67:22 - unused_field
   info - Cancel instances of dart.async.StreamSubscription - lib\player\player.dart:70:40 - cancel_subscriptions
   info - The value of the field '_stateChangeSubscription' isn't used - lib\player\player.dart:70:40 - unused_field
   info - Cancel instances of dart.async.StreamSubscription - lib\player\player.dart:73:28 - cancel_subscriptions
   info - The value of the field '_completionSubscription' isn't used - lib\player\player.dart:73:28 - unused_field
   info - Cancel instances of dart.async.StreamSubscription - lib\player\player.dart:76:30 - cancel_subscriptions
   info - The value of the field '_playerErrorSubscription' isn't used - lib\player\player.dart:76:30 - unused_field
   info - This function has a return type of 'Future<Duration>', but doesn't end with a return statement -
          lib\player\player.dart:160:3 - missing_return
   info - The method '_isIntentActionView' isn't used - lib\player\player.dart:326:16 - unused_element
   info - The value of the local variable 'result' isn't used - lib\player\player.dart:470:15 - unused_local_variable
   info - The value of the local variable 'result' isn't used - lib\player\player.dart:476:15 - unused_local_variable
   info - The for, if and spread elements were not supported until version 2.2.2, but this code is required to be able
          to run on earlier versions - lib\player\player.dart:691:17 - sdk_version_ui_as_code
   info - The value of the local variable 'permissions' isn't used - lib\player\player.dart:728:44 -
          unused_local_variable
   info - Unused import: 'package:app/heroes/albumArtHero.dart' - lib\routes\playerRoute.dart:5:8 - unused_import

22 issues found. (ran in 262.7s)
```

### `flutter doctor -v`

```shell
[flutter] flutter doctor -v
[√] Flutter (Channel unknown, v1.7.8+hotfix.4, on Microsoft Windows [Version 10.0.18362.356], locale ru-RU)
    • Flutter version 1.7.8+hotfix.4 at C:\dev\src\flutter
    • Framework revision 20e59316b8 (9 weeks ago), 2019-07-18 20:04:33 -0700
    • Engine revision fee001c93f
    • Dart version 2.4.0

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.0-rc1)
    • Android SDK at C:\Users\danya\AppData\Local\Android\Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.0-rc1
    • ANDROID_HOME = C:\Users\danya\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    X Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit https://flutter.dev/setup/#android-setup for detailed instructions.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 38.2.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] VS Code (version 1.38.1)
    • VS Code at C:\Users\danya\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.4.1

[√] Connected device (1 available)
    • Redmi Note 5 • 40c9f14 • android-arm64 • Android 9 (API 28)

! Doctor found issues in 1 category.
exit code 0
```

**Comment by nt4f04uNd** [comment]
The original issue resolved when I accidentally converted the stateless widget where list view had been used to statefull. But now I want to use that widget on same route but in tabs, and performance still actually very bad because of the massive freeze that occurs when I switch to tab where `ListView` is located

**Comment by nt4f04uNd** [comment]
The issue resolved when I moved `ListView` to `StatefulWidget` from `StatelessWidget`

**Comment by liyuqian** [comment]
> The issue resolved when I moved `ListView` to `StatefulWidget` from `StatelessWidget`

CC @filiph : I haven't looked into the details of this app, but it might be a nice example of real performance issues that are solved by just tweaking the app code.

**Comment by nt4f04uNd** [comment]
@liyuqian by the way I think I mistook in the reason of that. I found the [commit](https://github.com/nt4f04uNd/sweyer/commit/1a0cf31d8ee46aca78d45c69484a6599572fefe6) that solved this and looked upon the code changes. `lib/components/track_list.dart` - `TrackList` class: the `PageStorageKey` was declared just as `final` field of `StatelessWidget`, but after changes I made it `static final` field of `StatefulWidget`. Not sure, but I guess the true solution actually was declaring it as static 🤷‍♂️
